components/gitpod-protocol/src/util/analytics.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import Analytics = require("analytics-node");
+import { log } from './logging';
+
+export const IAnalyticsWriter = Symbol("IAnalyticsWriter");
+
+type Identity =
+    | { userId: string | number }
```

Comment:
why number?
why not `{ user?: { id: string }}` and handle the plumbing internally, so we save the `userId: user?.id || ""`

Reply:
to make this shape match segment